### PR TITLE
Preper For Shooting

### PR DESCRIPTION
**Pull request by DrorMargalit17**
adding interpoilation to ShooterArm, creating Paralall command group (prepertion for shooting) and StartFeeder command if ShooterArm and Shooter ready

### CODE REVIEW

File: src/main/java/frc/robot/commands/feederCommands/StartFeeder.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import frc.robot.subsystems.ShooterSubsysem;
+
+public class StartFeeder extends Command {
+  private final ShooterSubsysem shooterSubsysem = ShooterSubsysem.getInstance();
+  private final ShooterArmSubsystem shooterArmSubsystem = ShooterArmSubsystem.getInstance();
+  private final FeederSubsystem feederSubsystem = FeederSubsystem.getInstance();
+  double speed;
+  /** Creates a new StartFeeder. */
+  public StartFeeder(double speed) {
+    this.addRequirements(shooterSubsysem);
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/feederCommands/StartFeeder.java]: shooter isnt a requirements feeder is

File: src/main/java/frc/robot/commands/feederCommands/StartFeeder.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/feederCommands/StartFeeder.java:1]: change the name to feed to shooter

File: src/main/java/frc/robot/commands/feederCommands/StartFeeder.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import frc.robot.subsystems.ShooterArmSubsystem;
+import frc.robot.subsystems.ShooterSubsysem;
+
+public class StartFeeder extends Command {
+  private final ShooterSubsysem shooterSubsysem = ShooterSubsysem.getInstance();
+  private final ShooterArmSubsystem shooterArmSubsystem = ShooterArmSubsystem.getInstance();
+  private final FeederSubsystem feederSubsystem = FeederSubsystem.getInstance();
+  double speed;
+  /** Creates a new StartFeeder. */
+  public StartFeeder(double speed) {
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/feederCommands/StartFeeder.java]: dont get a speed use a constant

File: src/main/java/frc/robot/commands/PreperForShootting.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/PreperForShootting.java:1]: this is not how you write prepare

File: src/main/java/frc/robot/commands/PreperForShootting.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+public class PreperForShootting extends ParallelCommandGroup {
+  private final ShooterSubsysem shooterSubsysem = ShooterSubsysem.getInstance();
+  double degree;
+  /** Creates a new PreperForShootting. */
+  public PreperForShootting(double degree) {
+    this.addRequirements(shooterSubsysem);;
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/PreperForShootting.java]: shooter arm is also a requirment

**Review by ori-coval — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/PreperForShootting.java (hunk 3 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/PreperForShootting.java]: fixed

File: src/main/java/frc/robot/commands/PreperForShootting.java
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/PreperForShootting.java:1]: fixed

File: src/main/java/frc/robot/commands/feederCommands/StartFeeder.java (hunk 1 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/feederCommands/StartFeeder.java]: fixed

File: src/main/java/frc/robot/commands/feederCommands/StartFeeder.java
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/feederCommands/StartFeeder.java:1]: fixed

File: src/main/java/frc/robot/commands/feederCommands/StartFeeder.java (hunk 2 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/feederCommands/StartFeeder.java]: fixed

File: src/main/java/frc/robot/commands/feederCommands/FeedToShooter.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import static frc.robot.Constants.FeederConstants.*;
+
+
+public class FeedToShooter extends Command {
+  private final ShooterSubsysem shooterSubsysem = ShooterSubsysem.getInstance();
+  private final ShooterArmSubsystem shooterArmSubsystem = ShooterArmSubsystem.getInstance();
+  private final FeederSubsystem feederSubsystem = FeederSubsystem.getInstance();
+  /** Creates a new StartFeeder. */
+  public FeedToShooter() {
+    this.addRequirements(feederSubsystem, shooterArmSubsystem, shooterSubsysem);
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/feederCommands/FeedToShooter.java]: only feederSubsystem is a requirement

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -306,8 +314,10 @@ public static class CollectingConstants{
     }
   public static class FeederConstants{
 
+    public static final int speed = 20;
```
Comment (ori-coval) [inline src/main/java/frc/robot/Constants.java]: `FeederShootSpeed`

File: src/main/java/frc/robot/commands/PrepareForShooting.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+public class PrepareForShooting extends ParallelCommandGroup {
+  private final ShooterSubsysem shooterSubsysem = ShooterSubsysem.getInstance();
+  private final ShooterArmSubsystem shooterArmSubsystem = ShooterArmSubsystem.getInstance();
+  double degree;
+  /** Creates a new PreperForShootting. */
+  public PrepareForShooting(double degree) {
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/PrepareForShooting.java]: remove `double degree` from both places

File: src/main/java/frc/robot/commands/PrepareForShooting.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    this.addRequirements(shooterSubsysem, shooterArmSubsystem);
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/PrepareForShooting.java]: no addRequirements for command Groups

**Review by ori-coval — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/PrepareForShooting.java (hunk 7 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/PrepareForShooting.java]: fixed

File: src/main/java/frc/robot/commands/PrepareForShooting.java (hunk 6 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/PrepareForShooting.java]: fixed

File: src/main/java/frc/robot/Constants.java (hunk 5 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/Constants.java]: fixed

File: src/main/java/frc/robot/commands/feederCommands/FeedToShooter.java (hunk 4 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/feederCommands/FeedToShooter.java]: fixed

**Review by ori-coval — APPROVED** [review]

**Pull request merged**